src/DominantJuice.sol

```diff
@@ -0,0 +1,399 @@
+
+        // Calculates the refund bonus, using PRBMath, based on the time from cycle start and the size of the pledge
+        uint256 pledgerRefundBonus = unwrap(
+            mul(div(pledgers.pledgerWeight[redeemer], pledgers.totalPledgeWeight), wrap(campaign.totalRefundBonus))
+        );
+
+        // Sanity Check
+        if (address(this).balance < pledgerRefundBonus) revert InsufficientFunds();
+
+        // This redeeem function is now locked for the current redeeming address.
+        pledgers.hasBeenRefunded[redeemer] = true;
+
+        // Sending early pledger refund bonus.
+        (bool sendSuccess,) = redeemer.call{value: pledgerRefundBonus}("");
+        require(sendSuccess, "Failed to send refund bonus.");
+
+        emit CycleRefundBonusWithdrawal(redeemer, pledgerRefundBonus);
+    }
+
+    /**
+     * @notice This function is callable after cycleExpiryDate if goal is met. However, for
+     * contingency, where a redeemer is unable to retrieve funds, after a locked period, the
+     * Campaign Manager can withdraw funds for disbursement to remaining redeemers . The unlock
+     * time is hardcoded for two weeks after the cycleExpiryDate to give pledgers confidence
+     * that funds are programmatically guaranteed to remain in the contract for that time.
+     * @param receivingAddress Campaign Manager must call this function but can withdraw funds to
+     * another address if desired.
+     */
+    function creatorWithdraw(address payable receivingAddress, uint256 amount)
+        external
+        payable
+        onlyRole(CAMPAIGN_MANAGER_ROLE)
+    {
+        uint256 unlockTime = campaign.cycleExpiryDate + 1209600; // Two weeks after expiry date
+        require(
+            (hasCycleExpired() && isTargetMet()) || ((block.timestamp > unlockTime) && !isTargetMet()),
+            "Cycle must be expired and successful, or it must be passed the lock period."
+        );
+        if (amount > address(this).balance) revert InsufficientFunds();
+
+        (bool success,) = receivingAddress.call{value: amount}("");
+        require(success, "Failed to withdraw cycle funds.");
+
+        emit CreatorWithdrawal(receivingAddress, amount);
+    }
+
+    /// Getters and Helpers
+
+    function getCampaignInfo() public view returns (Campaign memory) {
+        return campaign;
+    }
+
+    function getBalance() public view returns (uint256) {
+        return address(this).balance;
+    }
+
+    function isTargetMet() public view returns (bool) {
+        if (pledgers.totalAmountPledged >= campaign.cycleTarget) return true;
+        else return false;
+    }
+
+    function hasCycleExpired() public view returns (bool) {
+        if (block.timestamp >= campaign.cycleExpiryDate) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    function getCycleFundingStatus() public view returns (FundingStatus memory) {
+        FundingStatus memory status;
```

Comment:
can do this all in one more readable statement (in my opinion) with eg:
```
return FundingStatus({
  totalPledged: pledgers.totalAmountPledged,
  ...
});
```

Check out https://jeancvllr.medium.com/solidity-tutorial-all-about-structs-b3e7ca398b1e

Reply:
Will do.